### PR TITLE
Process to merge release branch back to develop

### PR DESCRIPTION
**Pull request by AnnKatF**
**Why**
The process to review the PR that merges the release branch to develop needed more clarity.
Most of the times this merge will trigger conflicts between both branches and the release engineer handling the PR might not have all the knowledge to fix them.

**How**
The PR for merging release back to develop should have as reviewers all the engineers that contributed to the release branch. This way everyone involved in the PRs that were merged to release will have to review and help fixing the conflicts when required.

### CODE REVIEW

File: Cookbook/Technical-Documents/ReleaseProcess.md [hunk 1]
```diff
@@ -78,7 +78,10 @@ There are usually two release engineers working at any given time. It goes witho
    * Make sure you create separate tags (and GitHub releases) for each app released on the AppStore (eg. Babylon 4.1.0 and Telus 4.1.0 would each have their own `babylon/4.1.0` and `telus/4.1.0` tags)
    * Set the body of the GitHub release to the content of the Release Notes for the app
    * Attach the zipped `xcarchive` as an artefact to the GitHub release (if you're using the automated release command, you can find the `*.xcarchive.zip` in the Artifacts top section in the CI build).
-1. Merge the changes back to develop.
+1. Merge release branch back to develop:
+	* Open a PR from release branch targeting develop.
```
Comment (j531) [inline Cookbook/Technical-Documents/ReleaseProcess.md]:
```suggestion
	* Open a PR from `release` branch targeting `develop`. 
```

File: Cookbook/Technical-Documents/ReleaseProcess.md [hunk 2]
```diff
@@ -78,7 +78,10 @@ There are usually two release engineers working at any given time. It goes witho
    * Make sure you create separate tags (and GitHub releases) for each app released on the AppStore (eg. Babylon 4.1.0 and Telus 4.1.0 would each have their own `babylon/4.1.0` and `telus/4.1.0` tags)
    * Set the body of the GitHub release to the content of the Release Notes for the app
    * Attach the zipped `xcarchive` as an artefact to the GitHub release (if you're using the automated release command, you can find the `*.xcarchive.zip` in the Artifacts top section in the CI build).
-1. Merge the changes back to develop.
+1. Merge release branch back to develop:
```
Comment (j531) [inline Cookbook/Technical-Documents/ReleaseProcess.md]:
```suggestion
1. Merge `release` branch back to `develop`:
```

File: Cookbook/Technical-Documents/ReleaseProcess.md [hunk 3]
```diff
@@ -78,7 +78,10 @@ There are usually two release engineers working at any given time. It goes witho
...
+	* Open a PR from release branch targeting develop. 
+	* Set as reviewers all the engineers who contributed to the release branch.
```
Comment (j531) [inline Cookbook/Technical-Documents/ReleaseProcess.md]:
```suggestion
	* Set as reviewers all the engineers who contributed to the `release` branch.
```

File: Cookbook/Technical-Documents/ReleaseProcess.md [hunk 4]
```diff
@@ -78,7 +78,10 @@ There are usually two release engineers working at any given time. It goes witho
...
+	* Set the Merge label once all the required reviewers have approved it.
```
Comment (j531) [inline Cookbook/Technical-Documents/ReleaseProcess.md]:
```suggestion
	* Set the _Merge_ label once all the required reviewers have approved it.
```

**Review by j531 — APPROVED** [review]
Sounds like a sensible change to me. Added some suggestions for formatting - feel free to ignore.

**Comment by AnnKatF** [comment]
Thanks for your review @j531 !

**Review by mr-v — APPROVED** [review]

File: Cookbook/Technical-Documents/ReleaseProcess.md [hunk 5]
```diff
@@ -78,7 +78,10 @@ There are usually two release engineers working at any given time. It goes witho
    * Make sure you create separate tags (and GitHub releases) for each app released on the AppStore (eg. Babylon 4.1.0 and Telus 4.1.0 would each have their own `babylon/4.1.0` and `telus/4.1.0` tags)
    * Set the body of the GitHub release to the content of the Release Notes for the app
    * Attach the zipped `xcarchive` as an artefact to the GitHub release (if you're using the automated release command, you can find the `*.xcarchive.zip` in the Artifacts top section in the CI build).
-1. Merge the changes back to develop.
+1. Merge `release` branch back to `develop`:
+	* Open a PR from `release` branch targeting `develop`.
```
Comment (mr-v) [inline Cookbook/Technical-Documents/ReleaseProcess.md]: Not sure what's the current status, but it used be the case that lane for submitting release build would already create PR. Maybe @ilyapuchka has up to date info?

Comment (ilyapuchka) [reply Cookbook/Technical-Documents/ReleaseProcess.md]: yes, it should be created automatically when we first bump app version on release branch. it's release branch that needs to be created manually ATM

**Review by diegopetrucci — CHANGES_REQUESTED** [review]
Please address @mr-v's comment.

File: Cookbook/Technical-Documents/ReleaseProcess.md [hunk 6]
```diff
@@ -78,7 +78,10 @@ There are usually two release engineers working at any given time. It goes witho
    * Make sure you create separate tags (and GitHub releases) for each app released on the AppStore (eg. Babylon 4.1.0 and Telus 4.1.0 would each have their own `babylon/4.1.0` and `telus/4.1.0` tags)
    * Set the body of the GitHub release to the content of the Release Notes for the app
    * Attach the zipped `xcarchive` as an artefact to the GitHub release (if you're using the automated release command, you can find the `*.xcarchive.zip` in the Artifacts top section in the CI build).
-1. Merge the changes back to develop.
+1. Merge `release` branch back to `develop`:
+	* Open a PR from `release` branch targeting `develop`. 
+	* Set as reviewers all the engineers who contributed to the `release` branch.
```
Comment (Ben-Henshall) [inline Cookbook/Technical-Documents/ReleaseProcess.md]: Might be worth noting to remove PullAssigners assigned reviewers if they haven't contributed

Comment (diegopetrucci) [reply Cookbook/Technical-Documents/ReleaseProcess.md]: ^ @AnnKatF

File: Cookbook/Technical-Documents/ReleaseProcess.md [hunk 7]
```diff
@@ -78,8 +78,12 @@ There are usually two release engineers working at any given time. It goes witho
    * Make sure you create separate tags (and GitHub releases) for each app released on the AppStore (eg. Babylon 4.1.0 and Telus 4.1.0 would each have their own `babylon/4.1.0` and `telus/4.1.0` tags)
    * Set the body of the GitHub release to the content of the Release Notes for the app
    * Attach the zipped `xcarchive` as an artefact to the GitHub release (if you're using the automated release command, you can find the `*.xcarchive.zip` in the Artifacts top section in the CI build).
-1. Merge `develop` into the release branch, and resolve the conflicts (if any)
-1. Add the `Merge` label to the PR created by the bot to merge the release branch into `develop`.
+1. Merge `release` branch back to `develop`:
+	* Open the Release PR ( PR from `release` branch targeting `develop`) which has been automatically created.
+  * Resolve the conflicts (if any)
+	* Set as reviewers all the engineers who contributed to the `release` branch.
```
Comment (diegopetrucci) [inline Cookbook/Technical-Documents/ReleaseProcess.md]:
```suggestion
	* Set as reviewers all the engineers who contributed to the `release` branch, and remove the ones automatically assigned by PullAssigners.
```

**Review by diegopetrucci — APPROVED** [review]
added one suggestions and then it's good to go

**Comment by ilyapuchka** [comment]
IMO we should not keep fixes in the release branch, no matter if they are blocking any work on develop (then ofc they have to be added to develop branch as well) or not. This way we already have every individual change reviewed and merged into develop and all the conflicts will be already addressed (unless ofc we add more changes to develop on top of these fixes). Then we simply don't need to review release branch merge, which is cumbersome for everyone because it has all the changes all together and so all the conflicts have to be resolved all together. Ideally when release is done the only change that should be in release PR is a version bump. I think if we keep syncing fixes between dev and release branch we can achieve that and it will be less work to do for release engineer (a bit more work for each dev, but they are already working on it anyway so I consider overhead not to be huge).
So I'd suggest for release branch fixes we should have two PRs, one pointing to release branch, one pointing to develop branch, which should be linked to each other and reviewed together.

**Comment by mluisbrown** [comment]
@ilyapuchka this approach would not work in all circumstances. Sometimes we have fixes in the release branch which are just for the release and would conflict with the "real" fix being done in develop.

**Comment by mluisbrown** [comment]
And also @ilyapuchka I'm not sure what you are proposing. That all fixes in the release branch should also be cherry picked back to `develop`? In a separate PR? That seems like a lot of procedural overhead for not much gain.

**Comment by AnnKatF** [comment]
I agree with @mluisbrown. I think we should keep the release bug fixing in the release branch. The release development cycle is independent of the normal development cycle and we shouldn't mix it. As we have been experiencing, there are cases where we just have a temporary fix for the release which won't be valid in develop because there is ongoing work that will replace it.
Besides, if we stick with only fixing blockers and criticals, the merge shouldn't become that difficult.

**Comment by AnnKatF** [comment]
However, @ilyapuchka this can be discussed in the next Banter. Changes in the process should involve the whole team. For now this document will reflect the current process.

**Pull request merged**